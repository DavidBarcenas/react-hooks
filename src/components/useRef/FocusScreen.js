import React, { useState } from 'react'
import { MultipleCustomHook } from '../useFetch/MultipleCustomHook'

export const FocusScreen = () => {
  const [show, setShow] = useState(false)

  return (
    <div className="text-center p-5">
      <h1 className="text-success">Focus Screen</h1>
      <hr/>
      { show && <MultipleCustomHook /> }
      <button className="btn btn-danger mt-2" onClick={() => setShow(!show)}>show / hide</button>
    </div>
  )
}
