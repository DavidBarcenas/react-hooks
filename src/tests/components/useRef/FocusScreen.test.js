import '@testing-library/jest-dom';
import React, { useState } from 'react'
import { shallow } from 'enzyme';
import { FocusScreen } from '../../../components/useRef/FocusScreen';

describe('<FocusScreen /> testing', () => {
  const wrapper = shallow(<FocusScreen />)
  
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
  
  
  test('should show MultipleCustomHook component', () => {
    wrapper.find('button').simulate('click')
    expect(wrapper.find('MultipleCustomHook').exists()).toBe(true)
  })

})
