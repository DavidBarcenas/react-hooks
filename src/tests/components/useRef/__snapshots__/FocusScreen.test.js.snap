// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FocusScreen /> testing should render correctly 1`] = `
<div
  className="text-center p-5"
>
  <h1
    className="text-success"
  >
    Focus Screen
  </h1>
  <hr />
  <button
    className="btn btn-danger mt-2"
    onClick={[Function]}
  >
    show / hide
  </button>
</div>
`;
